driver/conv_driver.hpp

```diff
@@ -3151,7 +3151,7 @@ int ConvDriver<Tgpu, Tref>::RunBackwardDataGPUReference()
         din_dev->FromGPU(GetStream(), din_host.data.data());
     else
     {
-        auto din_tmp = tensor<Tgpu>(miopen::deref(inputTensor).GetType());
```

Comment:
Good catch, talking out loud on what's happening here: 
 - On the right hand side, it is trying to create a tensor from the data type (`miopenDataType_t`). 
 - Looking at the constructor of the tensor, it "thought" that it is creating a new tensor of size of the enum value.

Reasoning this through I believe implicitly, it was doing an implicit cast from `miopenDataType_t` to `size_t`.  C++ does allow it to happen silently.

(Now, independent from your PR), to prevent this from happening, I think those enum type can be instead amended to `enum class` type. However, since those miopen type definitions happen from a C header, it is probably not possible... So nothing we can do to convert this to a compilation error.